ndk/src/font.rs

```diff
@@ -0,0 +1,401 @@
+            fmt,
+            "each byte in an axis tag must be in the range 0x20 to 0x7E"
+        )
+    }
+}
+
+impl std::error::Error for TryFromU32Error {}
+
+/// A native [`AFont *`]
+///
+/// [`AFont *`]: https://developer.android.com/ndk/reference/group/font#afont_close
+#[cfg(feature = "api-level-29")]
+#[derive(Debug)]
+pub struct Font {
+    ptr: NonNull<ffi::AFont>,
+}
+
+#[cfg(feature = "api-level-29")]
+impl Font {
+    /// Create an `Font` from a pointer
+    ///
+    /// # Safety
+    /// By calling this function, you assert that the pointer is a valid pointer to a native
+    /// `AFont`.
+    pub unsafe fn from_ptr(ptr: NonNull<ffi::AFont>) -> Self {
+        Self { ptr }
+    }
+
+    /// Returns the pointer to the native `AFont`.
+    pub fn ptr(&self) -> NonNull<ffi::AFont> {
+        self.ptr
+    }
+
+    /// Return a count of font variation settings associated with the current font.
+    ///
+    /// The font variation settings are provided as multiple tag-values pairs.
+    ///
+    /// For example, bold italic font may have following font variation settings: 'wght' 700,
+    /// 'slnt' -12. In this case, [`Font::axis_count`] returns 2 and [`Font::axis_tag_at`] and
+    /// [`Font::axis_value_at`] will return following values.
+    pub fn axis_count(&self) -> usize {
+        unsafe { ffi::AFont_getAxisCount(self.ptr.as_ptr()) }
+    }
+
+    /// Return an OpenType axis tag associated with the current font.
+    ///
+    /// See [`Font::axis_count`] for more details.
+    pub fn axis_tag_at(&self, idx: usize) -> AxisTag {
+        // Android returns Axis Tag in big-endian.
+        // See https://cs.android.com/android/platform/superproject/+/refs/heads/master:frameworks/base/native/android/system_fonts.cpp;l=197 for details
+        AxisTag(unsafe { ffi::AFont_getAxisTag(self.ptr.as_ptr(), idx as u32) })
+    }
+
+    /// Return an OpenType axis value associated with the current font.
+    ///
+    /// See [`Font::axis_count`] for more details.
+    pub fn axis_value_at(&self, idx: usize) -> f32 {
+        unsafe { ffi::AFont_getAxisValue(self.ptr.as_ptr(), idx as u32) }
+    }
+
+    /// Return a font collection index value associated with the current font.
+    ///
+    /// In case the target font file is a font collection (e.g. .ttc or .otc), this returns a non
+    /// negative value as an font offset in the collection. This always returns 0 if the target font
+    /// file is a regular font.
+    pub fn collection_index(&self) -> usize {
+        unsafe { ffi::AFont_getCollectionIndex(self.ptr.as_ptr()) }
+    }
+
+    /// Return an absolute path to the current font file.
+    ///
+    /// Here is a list of font formats returned by this method:
+    ///
+    /// * OpenType
+    /// * OpenType Font Collection
+    /// * TrueType
+    /// * TrueType Collection
+    ///
+    /// The file extension could be one of *.otf, *.ttf, *.otc or *.ttc.
+    /// The font file returned is guaranteed to be opened with `O_RDONLY`.
```

Comment:
Is it opened, or open**able**? I don't see a file descriptor (of an open file) here... only a path.

Reply:
I don't have deep knowledge of English grammar, but don't opened and openable mean the same thing in this case? How about "The font file specified by the returned path is ..."?

Reply:
`to be opened` AFAIK either implies that it has previously been opened and is still open when it is returned to you, or that "something somewhere else" validates that it will definitely be opened at some point?

It seems, but I am not sure, that neither is true here.

Reply:
To me, that sentence sounds like "It is guaranteed that the user can open the returned font file with `O_RDONLY`", I think "the user" part is dropped, and the sentence is written in the passive voice. If you feel confused, we can rewrite that part with the active voice like the following:
"It is guaranteed that the user can open the font file specified by the returned path with `O_RDONLY`."